### ISSUE TITLE
Add upstream's tests

### ISSUE THREAD

**Issue by Klaim** [build2-packaging/entt#1]
Currently the only test implemented will just try to build the Readme code from v3.3.0

Add tests from upstream/tests/... 
These tests require gtest (see https://github.com/build2-packaging/googletest we have a package but it lacks gmock support at the moment).

**Comment by Wmbat** [comment]
gtest is now packaged in cppget, we should explore this further !

**Comment by Klaim** [comment]
Going back to this:

- [x] Having `gtest` packaged was our initial requirement, it's now done so we can proceed.
- [ ] [splitting the package in two](https://github.com/build2-packaging/entt/issues/4) is the first thing to do to get the original tests running without adding `gtest` as dependency to the library (and some others as specified below).
- [ ] Then as a fist step we need to add the "simple basic tests", which are tests using [this macro in the upstream CMake for tests](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L90-L95); this also includes the examples.
   - [ ] **CAN BE IGNORED**: We will need to package the `cereal` serialization library as [some of these basic tests use it](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L163-L168)
- [ ] **CAN BE IGNORE** Then we can add the [library tests which are a bit more complicated](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L97-L102), they generate both a library and an executable linking to it. Once this is done we already have most of the coverage.
- [ ] **CAN BE IGNORED**Then there are [plugin case tests which are a bit more complex](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L104-L112) and probably requires some `testscript` scripting so that the plugin binary can be found at runtime when running the tests. Also note that for loading the plugins, [they download a copy of this `cr` one-header library](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L131-L136) which is currently not packaged, so...
   - [ ] ...we need to package `cr`: https://github.com/fungos/cr
 
 Edit: updated following the discussion
 
I will ignore the benchmarks for now.

- A special attention need to be done for some of the options/macros because they seem to be important, for examlpe the presence or not of `ENTT_HAS_HEADER_VERSION`;
- There is something that looks like [a basic ODR check in there that all the tests requires](https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt#L86-L88), not sure what this is about exactly but seems like it can be done easilly.

I believe that covers everything https://github.com/skypjack/entt/blob/v3.11.1/test/CMakeLists.txt , please tell me if I'm missing something @skypjack

**Comment by skypjack** [comment]
@Klaim I think we can ignore lib and cereal as well as the benchmark and a few others maybe?
These are _extra tests_ for specific purposes. However, the main testsuite covers more or less everything already.
Also consider that all the ones above are enabled or disabled via cmake using a few options. It's easy to _ignore_ them in case.

**Comment by Klaim** [comment]
@skyjpack Thanks for the info. Ok so we will focus on the normal tests, make them by default and just make a separate ticket for if someone wants to add more of the upstream tests.

We have a plan.
